File: src/test/java/de/lukaseichler/recurrentneuralnetwork/RecurrentNeuralNetworkTest.java
package de.lukaseichler.recurrentneuralnetwork;

import java.util.Arrays;
import org.testng.annotations.Test;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * @author leichler
 */
public class RecurrentNeuralNetworkTest {
    @Test
    public void calculate() throws Exception {
        Network network = new RecurrentNeuralNetwork();
        network.addLayer(2)
                .addLayer(1);
        double result = network.calculate(Arrays.asList(1.0, 2.0)).get(0);
        assertThat(network.calculate(Arrays.asList(1.0, 2.0)).get(0)).isNotEqualTo(result);
    }
}

File: src/test/scala/de/lukaseichler/recurrentneuralnetwork/RecurrentNeuralNetworkTest.scala
package de.lukaseichler.recurrentneuralnetwork

import org.scalatest.Matchers
import org.scalatest.testng.TestNGSuite

/**
 * @author Lukas Eichler
 */
class RecurrentNeuralNetworkTest extends TestNGSuite with Matchers {

  //@Test
  def calculate {
    val network: Network = new RecurrentNeuralNetwork
    network.addLayer(2).addLayer(2).addLayer(1)
    val result: Double = network.calculate(List(1.0, 2.0))(0)
    network.calculate(List(1.0, 2.0))(0) should not equal result
  }
}
